drivers/bt_ll_nrfxlib/clock_control/clock_control_ll_nrfxlib.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <soc.h>
+#include <errno.h>
+#include <device.h>
+#include <kernel_includes.h>
+#include <clock_control.h>
+#include <ble_controller_soc.h>
+#include <multithreading_lock.h>
+
+static int hf_clock_start(struct device *dev, clock_control_subsys_t sub_system)
+{
+	ARG_UNUSED(dev);
+
+	int errcode = MULTITHREADING_LOCK_ACQUIRE();
+
+	if (errcode == 0) {
+		errcode = ble_controller_hf_clock_request(NULL);
+		MULTITHREADING_LOCK_RELEASE();
+	}
+	if (errcode != 0) {
+		return -EFAULT;
+	}
+
+	bool blocking = POINTER_TO_UINT(sub_system);
+
+	if (blocking) {
+		bool is_running = false;
+
+		while (!is_running) {
+			errcode = MULTITHREADING_LOCK_ACQUIRE();
```

Comment:
Could we suspend the calling thread while waiting for the the clock?

Reply:
Can we optimize this later?

Reply:
how long does it take before is_running is true?

Reply:
I am less worried about power consumption but more about other threads starvation

Reply:
@thomasstenersen I think it could be optimized later. Can you create Jira issues for things spotted during this review that won't be fixed before merging the PR?

Reply:
I'm told 1,5ms, but depends a bit on the external crystal.

Reply:
1.5 ms is the longest supported, the dev-kits are around 400 us if I remember correctly.